modules/javafx.graphics/src/main/java/com/sun/glass/ui/Screen.java

```diff
@@ -370,7 +372,7 @@ public static void setEventHandler(EventHandler eh) {
      */
     public static void notifySettingsChanged() {
         // Save the old screens in order to dispose them later
-        List<Screen> oldScreens = screens;
+        Set<Screen> oldScreens = new HashSet<>();
```

Comment:
Does the order matter? I don't think so, but if it does, `LinkedHashSet` would preserve the order.

Reply:
No, it doesn't matter. 

The old screens are "disposed" just by setting their native pointer to 0L, so they can't be reused, while the new screens instances are passed to the windows, to keep an updated instance. 

Note that even if old screen and new screen have the very same information (nothing changed for this particular screen), since `WinWindow::notifyMoving` uses the equality operator (`screen 1 == screen 2`), we need to keep a valid instance in `Window::screen`, and therefore `Screen::dispose` is just a way of invalidating old instances. Then, since they are no longer referenced by any window, they can be gc'ed.

Reply:
I had exactly the same question, and looked into the implementation.
Is it likely the implementation would change and the order be important in the future?

Reply:
Agreed. In this case I don't see how it could ever matter.

I only raised the question because whenever I see an ordinary HashSet being iterated, I always ask myself whether the iteration order matters.

Reply:
> I only raised the question because whenever I see an ordinary HashSet being iterated, I always ask myself whether the iteration order matters.

I agree 100%, especially since bugs in that case are really hard to find and debug.

I remember from my personal experience tests that sometimes failed, because the test did call `iterator().next()` on a Collection, which was a HashSet.😄 

I just saw that I forgot to quote Andy on my previous answer. Just to emphasize this: I don't see how the implementation could be changed in such a way that the order would matter in the future.